OpenSim/Simulation/Model/Frame.h

```diff
@@ -27,39 +27,58 @@
 #include <OpenSim/Simulation/osimSimulationDLL.h>
 #include <OpenSim/Simulation/Model/ModelComponent.h>
 
-
-
 namespace OpenSim {
 
-class Model;
-
 //=============================================================================
 //=============================================================================
 /**
- * A Frame is a modeling abstraction that defines a Righthanded CoordinateSystem
- * to which components can be attached, such as markers. The Frame allows
- * for a convenient way to describe the location and orientation of such
- * components.
+ * A Frame is an OpenSim abstraction for a reference frame. It consists of 
+ * a right-handed set of three orthogonal axes. Frames are intended to provide
+ * convenient reference frames for locating phyical structures (such as joints
+ * and muscle attachments) as well as provide a convenient basis for performing
+ * spatial calculations such as a frame aligned with the normal direction of a
+ * contact surface and/or located at the center-of-pressure.
+ *
+ * The puprose of a Frame is to provide its Transform (translation of the origin
+ * and the orientation of its axes) in the Ground frame as a function of the
+ * Model's (SimTK::MultibodySystem's) state.
```

Comment:
(1) typo: puprose
(2) It doesn't seem right to call this the "purpose" of a Frame. I might say the purpose of a Frame object is to represent the concept of a coordinate frame or something like that. But here I think you might mean something like "Every Frame is capable of providing its orientation and location with respect to Ground" an not necessarily talk about "purpose" here at all.